src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -104,23 +106,26 @@ public static class TagHelperOutputExtensions
                 throw new ArgumentNullException(nameof(tagBuilder));
             }
 
-            foreach (var attribute in tagBuilder.Attributes)
+            foreach (var attribute in tagBuilder.AttributeValues)
             {
                 if (!tagHelperOutput.Attributes.ContainsName(attribute.Key))
                 {
-                    tagHelperOutput.Attributes.Add(attribute.Key, attribute.Value);
+                    tagHelperOutput.Attributes.Add(attribute.Key, new StringValuesTutuContent(attribute.Value));
                 }
                 else if (attribute.Key.Equals("class", StringComparison.OrdinalIgnoreCase))
                 {
                     TagHelperAttribute classAttribute;
 
                     if (tagHelperOutput.Attributes.TryGetAttribute("class", out classAttribute))
                     {
-                        tagHelperOutput.Attributes.SetAttribute("class", classAttribute.Value + " " + attribute.Value);
+                        // If everything were a string, writes (classAttribute.Value + " " + attribute.Value).
```

Comment:
Odd that classes are added in different orders in `TagHelperOutputExtensions` and `TagBuilder`
- `TagHelperOutputExtensions` prepends new classes while `TagBuilder` appends them
- semantically the same but should we be consistent?